tls/s2n_server_new_session_ticket.c

```diff
@@ -253,3 +253,58 @@ int s2n_tls13_server_nst_write(struct s2n_connection *conn, struct s2n_stuffer *
 
     return S2N_SUCCESS;
 }
+
+int s2n_tls13_server_nst_read(struct s2n_connection *conn, struct s2n_stuffer *input)
+{
+    notnull_check(conn);
+    notnull_check(input);
+
+    uint32_t ticket_lifetime = 0;
+    GUARD(s2n_stuffer_read_uint32(input, &ticket_lifetime));
+
+    struct s2n_ticket_fields ticket_fields = { 0 };
+    GUARD(s2n_stuffer_read_uint32(input, &ticket_fields.ticket_age_add));
+
+    uint8_t ticket_nonce_len = 0;
+    struct s2n_blob nonce = { 0 };
+    GUARD(s2n_stuffer_read_uint8(input, &ticket_nonce_len));
+    uint8_t nonce_data[UINT8_MAX] = { 0 };
+    GUARD(s2n_blob_init(&nonce, nonce_data, ticket_nonce_len));
+    GUARD(s2n_stuffer_read_bytes(input, nonce.data, ticket_nonce_len));
+
+    uint16_t session_ticket_len = 0;
+    GUARD(s2n_stuffer_read_uint16(input, &session_ticket_len));
+    POSIX_ENSURE(session_ticket_len > 0, S2N_ERR_SAFETY);
+    GUARD(s2n_realloc(&conn->client_ticket, session_ticket_len));
+    GUARD(s2n_stuffer_read(input, &conn->client_ticket));
+
+    if(conn->config->session_ticket_cb != NULL) {
+
+        /* Derive individual session ticket secret */
+        s2n_tls13_connection_keys(secrets, conn);
+        struct s2n_blob master_secret = { 0 };
+        GUARD(s2n_blob_init(&master_secret, conn->resumption_master_secret, sizeof(conn->resumption_master_secret)));
+        uint8_t session_secret_data[S2N_TLS13_SECRET_MAX_LEN] = { 0 };
+        GUARD(s2n_blob_init(&ticket_fields.session_secret, session_secret_data, secrets.size));
+        POSIX_GUARD_RESULT(s2n_tls13_derive_session_ticket_secret(&secrets, &master_secret, &nonce, &ticket_fields.session_secret));
```

Comment:
You'll want a test that verifies that you're calculating the session ticket secret correctly-- what if your inputs are wrong? I suggest a known value test using https://www.rfc-editor.org/rfc/rfc8448.html.